### PR TITLE
Improve characters#new feature specs

### LINKED ISSUE #1093: Test that invalid character preserves inputs
https://github.com/Marri/glowfic/blob/ad00696b4d2eeb0a3e19aec9fe6adf270a45fd79/spec/features/characters/new_spec.rb#L4-L36

### PR DESCRIPTION
**Pull request by teceler**
Fixes #1093 

~~I encountered some problems around icon selection and template selection b/c the js isn't getting loaded; fixing that may require #1146~~ (fixed)

### CODE REVIEW

**Comment by Throne3d** [comment]
Can you leave an explicit TODO in the codebase around the things we currently have commented out? That way we know what the comments are for directly.

File: spec/features/characters/new_spec.rb [hunk 1]
```diff
@@ -50,7 +69,62 @@
...
+    expect(page).to have_no_selector('.flash.error')
+    expect(page).to have_selector('.flash.success')
+    within('.flash.success') do
+      expect(page).to have_text('Character saved successfully.')
+    end
+
+    within('.character-info-box') do
+      expect(page).to have_selector('.character-name', text: 'Example character')
+      expect(page).to have_selector('.character-screenname', text: 'example_screenname')
+      expect(page).to have_selector('.character-icon')
```
Comment (Throne3d) [inline spec/features/characters/new_spec.rb]: Can you check that this character-icon has the correct alt text? Just to be sure it used the clicked icon when producing the character.

File: spec/features/characters/new_spec.rb [hunk 2]
```diff
@@ -50,7 +69,62 @@
     end
   end
 
-  scenario "Creating character with icon, description and extant template"
+  scenario "Creating character with icon, description and extant template", js: true do
+    user = login
+    icon = create(:icon, user: user, keyword: 'Example icon')
```
Comment (Throne3d) [inline spec/features/characters/new_spec.rb]: Like in the previous test, can we create another couple of random icons here to be sure it selects the correct one?

File: spec/features/characters/new_spec.rb [hunk 3]
```diff
@@ -50,7 +69,62 @@
...
+    template = create(:template, user: user, name: 'Example template')
```
Comment (Throne3d) [inline spec/features/characters/new_spec.rb]: Can we make another template to be sure it selected this intentionally rather than randomly?

**Comment by Throne3d** [comment]
I like these tests! My comments are just to be surer that this is testing correct behavior, instead of accidentally-correct behavior, but it's a significant improvement! :D

File: spec/features/characters/new_spec.rb (hunk 1 above)
Comment (teceler) [reply spec/features/characters/new_spec.rb]: I think checking the id from the link should do that already, actually, but now it checks the alt text too I guess?? xD

**Pull request merged**